blockonomics.php

```diff
@@ -204,16 +216,24 @@ public function installDB()
         //Generate callback secret
         $secret = md5(uniqid(rand(), true));
         Configuration::updateValue('BLOCKONOMICS_CALLBACK_SECRET', $secret);
+        $callback_url = Tools::getHttpHost(true, true) .
+            __PS_BASE_URI__ .
+            'modules/' .
+            $this->name .
+            '/callback.php?secret=' .
+            $secret;
         Configuration::updateValue(
             'BLOCKONOMICS_CALLBACK_URL',
-            Tools::getHttpHost(true, true) .
-                __PS_BASE_URI__ .
-                'modules/' .
-                $this->name .
-                '/callback.php?secret=' .
-                $secret
+            $callback_url
         );
 
+        // Setup temp wallet
+        $response = $this->getTempApiKey($callback_url);
```

Comment:
This is unnecessary if an APIKey already exists (plugin upgrade)
Reference: https://github.com/blockonomics/woocommerce-plugin/blob/bf647c58484cfe8837ff61aaf504303659c82e8c/blockonomics-woocommerce.php#L111

Reply:
This should not run on plugin upgrades, and the secret is also created in this section. 
The check for API has been added, however may be unnecessary.
04589e3

Reply:
Agreed. I am revert this commit